libwallet/assets/btc/dex-wallet.go

```diff
@@ -175,79 +328,522 @@ func (dw *DEXWallet) SignTx(tx *wire.MsgTx) error {
+	}
+
+	blockHeight, err := dw.cl.GetBlockHeight(blockHash)
+	if err != nil {
+		return nil, false, err
+	}
+
+	confirmations := int64(-1)
+	mainchain = dw.blockIsMainchain(blockHash, blockHeight)
+	if mainchain {
+		confirmations = int64(confirms(blockHeight, tip.Height))
+	}
+
+	return &dexbtc.BlockHeader{
+		Hash:              hdr.BlockHash().String(),
+		Confirmations:     confirmations,
+		Height:            int64(blockHeight),
+		Time:              hdr.Timestamp.Unix(),
+		PreviousBlockHash: hdr.PrevBlock.String(),
+	}, mainchain, nil
+}
+
+// Part of dexbtc.Wallet interface.
+func (dw *DEXWallet) GetBestBlockHeader() (*dexbtc.BlockHeader, error) {
+	hash, err := dw.GetBestBlockHash()
+	if err != nil {
+		return nil, err
+	}
+	hdr, _, err := dw.GetBlockHeader(hash)
+	return hdr, err
 }
 
-func (dw *DEXWallet) Birthday() time.Time {
-	return dw.w.Manager.Birthday()
+// Part of dexbtc.Wallet interface.
+func (dw *DEXWallet) Connect(_ context.Context, _ *sync.WaitGroup) (err error) {
+	return nil
 }
 
-func (dw *DEXWallet) Peers() ([]*asset.WalletPeer, error) {
-	peers := dw.spvService.CS.Peers()
-	var walletPeers []*asset.WalletPeer
-	for i := range peers {
-		p := peers[i]
-		walletPeers = append(walletPeers, &asset.WalletPeer{
-			Addr:      p.Addr(),
-			Connected: p.Connected(),
-			Source:    asset.WalletDefault,
-		})
+// GetTxOut finds an unspent transaction output and its number of confirmations.
+// To match the behavior of the RPC method, even if an output is found, if it's
+// known to be spent, no *wire.TxOut and no error will be returned.
+// Part of dexbtc.Wallet interface.
+func (dw *DEXWallet) GetTxOut(txHash *chainhash.Hash, vout uint32, _ []byte, _ time.Time) (*wire.TxOut, uint32, error) {
+	// Check for a wallet transaction first
+	txDetails, err := dw.walletTransaction(txHash)
+	if err != nil && !errors.Is(err, dexbtc.WalletTransactionNotFound) {
+		return nil, 0, fmt.Errorf("walletTransaction error: %w", err)
+	}
+
+	if txDetails == nil {
+		return nil, 0, nil // wallet has not seen tx
+	}
+
+	spent, found := outputSpendStatus(txDetails, vout)
+	if !found || spent {
+		return nil, 0, nil
+	}
+
+	if len(txDetails.MsgTx.TxOut) <= int(vout) {
+		return nil, 0, fmt.Errorf("wallet transaction %s doesn't have enough outputs for vout %d", txHash, vout)
+	}
+
+	var confs uint32
+	if txDetails.Block.Height > 0 {
+		tip, err := dw.cl.BestBlock()
+		if err != nil {
+			return nil, 0, fmt.Errorf("BestBlock error: %v", err)
+		}
+		confs = uint32(confirms(txDetails.Block.Height, tip.Height))
+	}
+
+	msgTx := &txDetails.MsgTx
+	if len(msgTx.TxOut) <= int(vout) {
+		return nil, 0, fmt.Errorf("wallet transaction %s found, but not enough outputs for vout %d", txHash, vout)
+	}
+
+	return msgTx.TxOut[vout], confs, nil
+
+	// We don't really know if it's spent, so we'll need to scan.
+	// utxo, err := dw.scanFilters(txHash, vout, pkScript, startTime, blockHash)
+	// if err != nil {
+	// 	return nil, 0, err
+	// }
```

Comment:
Some problem with scanFilters?

Reply:
Not really, wanted to have a working implementation up before this.